cvat-sdk/cvat_sdk/pytorch/__init__.py

```diff
@@ -298,6 +299,94 @@ def __len__(self) -> int:
         return len(self._active_frame_indexes)
 
 
+class ProjectVisionDataset(torchvision.datasets.VisionDataset):
+    """
+    Represents a project on a CVAT server as a PyTorch Dataset.
+
+    The dataset contains one sample for each frame of each task in the project
+    (except for tasks that are filtered out - see the description of `task_filter`
+    in the constructor). The sequence of samples is formed by concatening sequences
+    of samples from all included tasks in an arbitrary order that's consistent
+    between executions. Each task's sequence of samples corresponds to the sequence
+    of frames on the server.
+
+    See `TaskVisionDataset` for information on sample format, caching, and
+    current limitations.
+    """
+
+    def __init__(
+        self,
+        client: cvat_sdk.core.Client,
+        project_id: int,
+        *,
+        transforms: Optional[Callable] = None,
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+        label_name_to_index: Mapping[str, int] = None,
+        task_filter: Optional[Callable[[models.ITaskRead], bool]] = None,
```

Comment:
Should we include a subset filter too? I understand that it can be expressed via a task filter, but it may be closer to what the user needs.

Reply:
I don't see much reason to do that. `task_filter=lambda t: t.subset == 'Train'` isn't much more complicated to write than `subset_filter=lambda s: s == 'Train'`.

Reply:
I meant something like a list with required subset names. As we've heard, users don't really want to learn SDK APIs, even though they are simple enough.

Reply:
I guess I can do that...

Reply:
Okay, implemented.